torch/quantization/_correct_bias.py

```diff
@@ -0,0 +1,121 @@
+    '''
+    current = model
+    name = name.split('.')
+    for subname in name:
+        if subname == '':
+            return current
+        current = current._modules[subname]
+    return current
+
+def get_parent_module(model, name):
+    ''' Given name of submodule, this function grabs the parent of the submodule, from given model
+    '''
+    parent_name = name.rsplit('.', 1)[0]
+    if parent_name == name:
+        parent_name = ''
+    return get_module(model, parent_name)
+
+def get_param(module, attr):
+    ''' Sometimes the weights/bias attribute gives you the raw tensor, but sometimes
+    gives a function that will give you the raw tensor, this function takes care of that logic
+    '''
+    param = getattr(module, attr, None)
+    if isinstance(param, nn.Parameter) or isinstance(param, torch.Tensor):
+        return param
+    elif callable(param):
+        return param()
+    return None
+
+class MeanShadowLogger(ns.Logger):
+    r"""A logger for a Shadow module whose purpose is to record the rolling mean
+    of the data passed to the floating point and quantized models
+    """
+    def __init__(self):
+        super(MeanShadowLogger, self).__init__()
+        self.stats["float"] = None
+        self.stats["quantized"] = None
+        self.count = 0
+        self.float_sum = None
+        self.quant_sum = None
+
+    def forward(self, x, y):
+        if len(x) > 1:
+            x = x[0]
+        if len(y) > 1:
+            y = y[0]
+        if x.is_quantized:
+            x = x.dequantize()
+
+        self.count += 1
+        if self.stats["quantized"] is None:
+            self.stats["quantized"] = x
+            self.quant_sum = x
+        else:
+            self.quant_sum += x
+            self.stats["quantized"] = self.quant_sum / self.count
+
+        if self.stats["float"] is None:
+            self.stats["float"] = y
+            self.float_sum = y
+        else:
+            self.float_sum += y
+            self.stats["float"] = self.float_sum / self.count
+
+def bias_correction(float_model, quantized_model, img_data, neval_batches=30):
+    ''' Using numeric suite shadow module, the expected output of the floating point and quantized modules
+    is recorded. Using that data the bias of supported modules is shifted to compensate for the drift caused
+    by quantization
+    Paper reference: https://arxiv.org/pdf/1906.04721.pdf (Section 4.2)
+    '''
+    biased_modules = {}
+    for name, submodule in quantized_model.named_modules():
+        if type(submodule) in _supported_modules_quantized:
+            biased_modules[name] = submodule
+
+    for biased_module in biased_modules:
+        quantized_submodule = get_module(quantized_model, biased_module)
+        bias = get_param(quantized_submodule, 'bias')
+        if bias is not None:
+            ns.prepare_model_with_stubs(float_model, quantized_model, _supported_modules, MeanShadowLogger)
```

Comment:
is float_model corresponding to `x` and quantizd_model corresponding to `y` in the forward function of `MeanShadowLogger`?